src/magic_folder/test/test_config.py

```diff
@@ -547,20 +548,51 @@ def setup_example(self):
 
     @given(
         remote_snapshots(),
+        path_states(),
     )
-    def test_remotesnapshot_roundtrips(self, snapshot):
+    def test_remotesnapshot_roundtrips(self, snapshot, path_state):
         """
         The capability for a ``RemoteSnapshot`` added with
         ``MagicFolderConfig.store_remotesnapshot`` can be read back with
         ``MagicFolderConfig.get_remotesnapshot``.
         """
-        self.db.store_remotesnapshot(snapshot.name, snapshot)
-        loaded = self.db.get_remotesnapshot(snapshot.name)
+        self.db.store_remotesnapshot(snapshot.name, snapshot, path_state)
+        capability = self.db.get_remotesnapshot(snapshot.name)
+        db_path_state = self.db.get_currentsnapshot_pathstate(snapshot.name)
         self.assertThat(
-            snapshot.capability,
-            Equals(loaded),
+            (capability, db_path_state),
+            Equals((snapshot.capability, path_state))
         )
 
+    @given(
+        remote_snapshots(),
+        path_states(),
+    )
+    def test_remotesnapshot_with_existing_state(self, snapshot, path_state):
+        """
+        A ``RemoveSnapshot`` can be add without path state, if existing path
+        state is in the database.
+        """
+        self.db.store_currentsnapshot_state(snapshot.name, path_state)
+        self.db.store_remotesnapshot(snapshot.name, snapshot, path_state)
+        capability = self.db.get_remotesnapshot(snapshot.name)
+        db_path_state = self.db.get_currentsnapshot_pathstate(snapshot.name)
+        self.assertThat(
+            (capability, db_path_state),
+            Equals((snapshot.capability, path_state))
+        )
+
+    @given(
+        remote_snapshots(),
+    )
+    def test_store_remote_without_state(self, snapshot):
+        """
+        Calling :py:`MagicFolderConfig.store_remotesnapshot` without a path
+        state, when there isn't already corresponding path state fails.
+        """
+        with ExpectedException(Exception):
```

Comment:
Maybe `store_remotesnapshot` can document this exceptional case.  It might also be nice to raise a distinctive exception type for it rather than using the generic Exception.